arborist/auth.go

```diff
@@ -76,3 +99,50 @@ func (requestJSON *AuthRequestJSON_Request) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+// Authorize the given token to access resources by service and method.
+// The given resource can be an expression of slash-separated resource paths
+// connected with `and`, `or` or `not`. The priority of these boolean operators
+// is: `not > and > or`. When in doubt, use parenthesises to specify explicitly.
+func authorize(server *Server, token *TokenInfo, resource string,service string, method string) (bool, error) {
```

Comment:
let's pass an `AuthRequest` argument instead of token + resource + service + method separately?

Reply:
I also like passing a struct response instead of `bool` so we can more easily add some error JSON if more detail is added to the responses

Reply:
Okay sure thing!